### PR TITLE
Add workflows with partial backends to CI

### PR DESCRIPTION
**Pull request by mgorny**
## What do these changes do?

Attempt to add two new CI workflows, one testing without memcached and the other without redis.

## Are there changes in behavior for the user?

No.

## Related issue number

Suggested in #725

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: `Fix issue with non-ascii contents in doctest text files.`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -111,6 +111,54 @@ jobs:
         flags: unit
         fail_ci_if_error: false
 
+  test-partial-backends:
```
Comment (Dreamsorcerer) [inline .github/workflows/ci.yml:114]: I wonder if it'd be simpler to just add an extra step to the existing test job, which will probably help to keep all the coverage steps together etc.

Comment (Dreamsorcerer) [reply .github/workflows/ci.yml:114]: I don't think a matrix will add much. We can just uninstall both libraries and do 1 extra test run.

**Comment by mgorny** [comment]
Turned out to be surprisingly easy. I went for removing them after installing to avoid changing `requirements.txt` since we still want them installed by default, and they require special pytest arguments anyway.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mgorny) [reply .github/workflows/ci.yml:114]: Sure, will update.

**Comment by mgorny** [comment]
Done. FWICS the 'else' clauses are now green in codecov, so I suppose it worked ;-).

**Comment by Dreamsorcerer** [comment]
Awesome, thanks.

**Pull request merged**